### PR TITLE
NPC creation

### LINKED ISSUE #20: NPC (Non player character creation)

### PR DESCRIPTION
**Pull request by sethtrain**
Just a simple endpoint to create NPCs

I would like @omnomicon to speak to the possible need for something like this to have a unique identifier (UUID) for easy updating.

Fixes #20

### CODE REVIEW

File: s-resources-cf.json [hunk 1]
```diff
@@ -64,6 +64,51 @@
...
+          }
+        ],
+        "GlobalSecondaryIndexes": [],
+        "KeySchema": [
+          {
+            "AttributeName": "game",
+            "KeyType": "HASH"
+          }
+        ],
+        "LocalSecondaryIndexes": [
```
Comment (omnomicon) [inline s-resources-cf.json]: i don't think you need this

File: s-resources-cf.json [hunk 2]
```diff
@@ -64,6 +64,51 @@
...
+            "AttributeName": "name",
+            "AttributeType": "S"
+          },
+          {
+            "AttributeName": "story",
+            "AttributeType": "S"
+          }
+        ],
+        "GlobalSecondaryIndexes": [],
+        "KeySchema": [
```
Comment (omnomicon) [inline s-resources-cf.json]: i'd add a sort key on `name`. that way you can select a particular npc with `game + name` or all npcs in a game with just `game`

File: s-resources-cf.json [hunk 3]
```diff
@@ -64,6 +64,51 @@
...
+          {
+            "AttributeName": "game",
+            "AttributeType": "S"
+          },
+          {
+            "AttributeName": "name",
+            "AttributeType": "S"
+          },
+          {
+            "AttributeName": "story",
```
Comment (omnomicon) [inline s-resources-cf.json]: you'll want to take this out since it's not part of the key

Comment (sethtrain) [inline s-resources-cf.json]: Wouldn't `name` go into the `KeySchema` (with the KeyType field added of course) and the `AttributeDefinitions` stay the same?

Comment (sethtrain) [inline s-resources-cf.json]: Nevermind, I get it now.

**Comment by omnomicon** [comment]
schema looks good! I was thinking since (other than keys) dynamo is schema-less, you could actually have separate columns for HP, XP, $, etc. if you're running a game where those don't apply (but other things do), then it will happily not add those columns, but add the new ones,

**Comment by sethtrain** [comment]
Yeah, that makes sense.  So basically the Ruleset would determine the "form" used to generate these values?

File: s-resources-cf.json [hunk 4]
```diff
@@ -64,6 +64,38 @@
...
+        "GlobalSecondaryIndexes": [],
+        "KeySchema": [
+          {
+            "AttributeName": "game",
+            "KeyType": "HASH"
+          },
+          {
+            "AttributeName": "name",
+            "KeyType": "RANGE"
+          },
```
Comment (omnomicon) [inline s-resources-cf.json]: bad comma

**Pull request merged**